torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -962,12 +958,13 @@ class CudaKernelGenerator : private kir::IrVisitor {
   }
 
   void visit(const kir::IfThenElse* node) final {
-    if (node->cond()->isConst() && node->cond()->value().value()) {
+    auto conditional = node->cond()->value();
```

Comment:
`Predicate` has both `value_` and `thread_pred_`. Is `thread_pred_` not relevant at this point?

Reply:
`thread_pred_` corresponds directly with the `thread_pred` argument for `PredicateCompute::getInlinePredicate`

Reply:
So, at this point, `node->cond()->predicate_type()` is guaranteed to be `PredicateType::Value`?

Reply:
I was going to keep the PredicateType constant. However, value should no longer be a nullptr.

Reply:
OK. It wasn't very clear to me how those multiple values of `Predicate` are supposed to be used.

Reply:
In hindsight, using a `PredicateType::Value` type and a value field is slightly confusing. Maybe `PredicateType::Manual` is a better choice.